x-pack/legacy/plugins/triggers_actions_ui/np_ready/public/application/components/builtin_action_types/email.tsx

```diff
@@ -0,0 +1,550 @@
+        defaultMessage: 'Configure settings to send email through your mail server',
+      }
+    ),
+    validateConnector: (action: ActionConnector): ValidationResult => {
+      const validationResult = { errors: {} };
+      const errors = {
+        from: new Array<string>(),
+        service: new Array<string>(),
+        port: new Array<string>(),
+        host: new Array<string>(),
+        user: new Array<string>(),
+        password: new Array<string>(),
+      };
+      validationResult.errors = errors;
+      if (!action.config.from) {
+        errors.from.push(
+          i18n.translate(
+            'xpack.triggersActionsUI.components.builtinActionTypes.error.requiredFromText',
+            {
+              defaultMessage: 'From is required.',
+            }
+          )
+        );
+      }
+      if (!action.config.port && !action.config.service) {
+        errors.port.push(
+          i18n.translate(
+            'xpack.triggersActionsUI.components.builtinActionTypes.error.requiredPortText',
+            {
+              defaultMessage: 'Port or Service is required.',
+            }
+          )
+        );
+      }
+      if (!action.config.service && (!action.config.port || !action.config.host)) {
+        errors.service.push(
+          i18n.translate(
+            'xpack.triggersActionsUI.components.builtinActionTypes.error.requiredServiceText',
+            {
+              defaultMessage: 'Service or host with port is required.',
+            }
+          )
+        );
+      }
+      if (!action.config.host && !action.config.service) {
+        errors.host.push(
+          i18n.translate(
+            'xpack.triggersActionsUI.components.builtinActionTypes.error.requiredHostText',
+            {
+              defaultMessage: 'Host or Service is required.',
+            }
+          )
+        );
+      }
+      if (!action.secrets.user) {
+        errors.user.push(
+          i18n.translate(
+            'xpack.triggersActionsUI.components.builtinActionTypes.error.requiredUserText',
+            {
+              defaultMessage: 'User is required.',
+            }
+          )
+        );
+      }
+      if (!action.secrets.password) {
+        errors.password.push(
+          i18n.translate(
+            'xpack.triggersActionsUI.components.builtinActionTypes.error.requiredPasswordText',
+            {
+              defaultMessage: 'Password is required.',
+            }
+          )
+        );
+      }
+      return validationResult;
+    },
+    validateParams: (actionParams: any): ValidationResult => {
+      const validationResult = { errors: {} };
+      const errors = {
+        to: new Array<string>(),
+        cc: new Array<string>(),
+        bcc: new Array<string>(),
+        message: new Array<string>(),
+        subject: new Array<string>(),
+      };
+      validationResult.errors = errors;
+      if (
+        (!actionParams.to || !(actionParams.to instanceof Array) || actionParams.to.length === 0) &&
```

Comment:
There's alot going on in this one `if`, and looks like we can actually omit a third of them because: `!actionParams.to || !(actionParams.to instanceof Array)` is essentially the same as `!(actionParams.to instanceof Array)` right?